### PR TITLE
sasjs doc

### LINKED ISSUE #368: sasjs doc
Given that SASjs enforces (or expects)  a doxygen-formatted header in the SAS Macros, it would be nice to embed a documentation generator for the SAS content.

Something like:

```
sasjs doc [ --outDirectory /some/location  -t sometarget]
```

We could expect a doxygen configuration file under `/sasjs/doxy/doxyfile` as part of `sasjs create` along with some custome header / footer (sasjs branded).

The reason to use `sasjs doc` over the `doxygen` command directly is that we will use SASjs to insert the paths to all the SAS programs / macros/ jobs / services from the sasjsconfig.json file

By default the output should go in the `sasjsbuild` folder, unless an alternative location is chosen.

If a target is supplied, we should generate docs only for the SAS programs / macros / jobs / services in that target (and the root).

If no target is supplied, we should generate for all sas programs/  macros / jobs / services.

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

Closes #368 

## Implementation

- Replaced `cp-cli` with `copyfiles` npm package, update package.json
- export `isWindows()` from _utils/command.ts_ to use it other places.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [ ] Any new functionality has been unit tested.
- [ ] All unit tests are passing (`npm test`).
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [ ] Reviewer is assigned.

## Directions to test

- checkout branch 'doxygen'
- `npm i && npm start`
- create new sasjs app & issue command `sasjs doc`

### CODE REVIEW

File: src/commands/docs/docs.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    `${doxyParams} doxygen ${doxyConfigPath}`,
+    {
+      silent: true
+    }
+  )
+  spinner.stop()
+
+  if (code !== 0) {
+    throw new Error(
+      `'Doxygen' is not installed or configured, make sure 'doxygen' avaiable in console.\n${stderr}`
```
Comment (allanbowe) [inline src/commands/docs/docs.ts:68]: How about:

`'The Doxygen application is not installed or configured.  This external tool is used by 'sasjs doc'.  Please download and install from here:  https://www.doxygen.nl/download.html#srcbin '`

File: src/commands/help/help.ts [hunk 2]
```diff
@@ -265,6 +265,16 @@ export async function printHelpText() {
         )} - triggers flow for execution.`,
         `[2spaces]command example: sasjs flow execute --source /local/flow.json --logFolder /local/log/folder --csvFile /local/some.csv`
       ]
+    },
+    {
+      name: 'doc',
+      title: 'doc <command>',
+      description: [
+        `Generates docs for SAS Programs / Macros / Jobs / Services for all targets and root of sasjsconfig by default. If target is provided, generated docs will be according to target.`,
```
Comment (allanbowe) [inline src/commands/help/help.ts:273]: `Generates docs for all SAS Programs / Macros / Jobs / Services listed in the sasjsconfig.json file by default. If a target is provided, the generated docs will ignore content from the other targets.`

File: src/commands/help/help.ts [hunk 3]
```diff
@@ -265,6 +265,16 @@ export async function printHelpText() {
...
+        `[2spaces]command example: sasjs doc -t <targetName> --outDirectory <sasFilePath>`,
+        `[2spaces]NOTE: Providing outDirectory flag is optional. If present, CLI will generate docs to outDirectory.`,
```
Comment (allanbowe) [inline src/commands/help/help.ts:275]: what happens if it is not present?

Comment (saadjutt01) [reply src/commands/help/help.ts:275]: default is _./sasjsbuild/docs_

**Review by allanbowe — APPROVED** [review]

**Pull request merged**